experimental/learn_descriptors/four_seasons_parser_detail.cc

```diff
@@ -0,0 +1,220 @@
+#include "experimental/learn_descriptors/four_seasons_parser_detail.hh"
+
+#include <chrono>
+#include <cmath>
+#include <cstddef>
+#include <fstream>
+#include <iostream>
+#include <optional>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "absl/strings/str_split.h"
+#include "absl/strings/strip.h"
+#include "nmea/message/gga.hpp"
+#include "nmea/message/rmc.hpp"
+#include "nmea/sentence.hpp"
+
+namespace robot::experimental::learn_descriptors::detail::four_seasons_parser {
+namespace txt_parser_help {
+
+std::vector<std::string> parse_line_adv(const std::string& line, const std::string& delim) {
+    if (delim == " ") {
+        return std::vector<std::string>(
+            absl::StrSplit(line, absl::ByAnyChar(" \t\n\r"), absl::SkipWhitespace()));
+    }
+    return std::vector<std::string>(absl::StrSplit(line, delim, absl::SkipWhitespace()));
+}
+
+std::shared_ptr<CameraCalibrationFisheye> load_camera_calibration(
```

Comment:
Any reason the return value couldn't be a `CameraCalibrationFisheye`? My understanding is that the calibration only needs to be shared between the parser and the image points, but if someone else wanted to load the camera calibrations, shouldn't be forced into using a shared pointer.